### PR TITLE
Deprecated torch < 2.0.0.

### PR DESCRIPTION
**Pull request by luisenp**
Bumps version to `0.2.0.dev0`

### CODE REVIEW

File: theseus/_version.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import warnings
+
+from torch import __version__ as _torch_version
+from semantic_version import Version
+
+if Version(_torch_version) < Version("2.0.0"):
+    warnings.warn(
+        "Using torch < 2.0 is deprecated and support will be discontinued "
```
Comment (mhmukadam) [inline theseus/_version.py:8]: will be discontinued -> is discontinued?

File: theseus/core/cost_function.py [hunk 2]
```diff
@@ -11,7 +11,11 @@
 
 import torch
 import torch.autograd.functional as autogradF
-from functorch import jacrev, vmap
+
+try:
+    from torch.func import jacrev, vmap
+except ModuleNotFoundError:
+    from functorch import jacrev, vmap  # type: ignore
```
Comment (mhmukadam) [inline theseus/core/cost_function.py:18]: Is this in case torch < 2.0 is installed and functorch is separate install+import?

**Review by mhmukadam — COMMENTED** [review]
LGTM

File: theseus/core/cost_function.py (hunk 2 above)
Comment (luisenp) [reply theseus/core/cost_function.py:18]: Yes!

File: theseus/_version.py (hunk 1 above)
Comment (luisenp) [reply theseus/_version.py:8]: Changed the message here 103219c2df6db8c924fcc16649faa0e5380b5402.

**Pull request merged**